R/error_checking.R

```diff
@@ -0,0 +1,340 @@
+#'
+#' This function runs a check on map data inputs and looks for errors
+#' @param map_data (Data Frame or Character) - A data frame that contains the output data to map, or alternatively a full path to a CSV
+#' @param data_key_field (Character) - Name of key field in data_obj for merging with shape_data
+#' @param data_col (Character) - Column name that contains the data object's output variable
+#' @return (Character) - Returns a success token or an error string if failed
+#' @export
+verify_data <- function(map_data, data_key_field, data_col)
+{
+    # Future
+    # 	Does it have nulls?
+    #  	Warning that pops up to the user about null values
+
+    result <- "Success"
+
+    # Verify map_data
+    if(!is.null(map_data))
+    {
+        # Verify data_key_field
+        if(!is.null(data_key_field))
+        {
+            if(!"character" %in% class(data_key_field))
+                return("Error: data_key_field argument must be of type character")
+            if(!data_key_field %in% colnames(map_data))
+                return("Error: data_key_field was not found in the map_data data frame")
+        }
+
+        # Verify data_col
+        if(!is.null(data_col))
+        {
+            if(!"character" %in% class(data_col))
+                return("Error: data_col argument must be of type character")
+            if(!data_col %in% colnames(map_data))
+                return("Error: data_col was not found in the map_data data frame")
+        }
+    }
+
+    return(result)
+}
+
+
+#' Verify csv data
+#'
+#' This function runs a check on csv data inputs and looks for errors
+#' @param map_data (Data Frame or Character) - Either the full path string to data file (csv) or a data.frame object
+#' @return (Character) - Returns a success token or an error string if failed
+#' @export
+verify_csv <- function(map_data)
+{
+
+    # Check csv for things like character Unicode
+    # Check field they provided there etc
+
+
+    result <- "Success"
+
+    return(result)
+
+}
+
+
+#' Verify map parameters
+#'
+#' @param bin_method (Character) - Method or function to use to split continuous data into discrete chunks (one of "quantile", "equal", "pretty", "kmeans") (default "pretty")
+#' @param bins (Numeric) - Number of bins/segments in which to divide the raster
+#' @param dpi (Numeric) - Settable DPI for different print/screen formats (default 150)
+#' @param expand_xy (c(Numeric, Numeric)) - Sets expansion amount for the X and Y scale of the map - Vector (expand x, expand y) (default c(0,0))
+#' @param map_xy_min_max (c(Numeric, ...)) - Vector that describes the desired extent of the map in the form of (xmin, xmax, ymin, ymax) (default: c(-180, 180, -90, 90))
+#' @param map_title (Character) - Title to be displayed on the output map
+#' @param map_palette (Character) - Optional variable to manually set the colorscale to a specific palette from RColorbrewer
+#' @param map_palette_reverse (Boolean) - Set palette to reverse direction TRUE/FALSE
```

Comment:
try not to use `/` this could cause problems with unicode

Reply:
Removed all '/' characters in this commit f520ed0cd80f23ce3cf066ba94018d9a33f3fd2f. I did not remove commented lines that contained a / in the formation of a URL. Should I remove any URL references too?